### PR TITLE
refactor: merge standalone and metasrv table metadata allocators

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

The two are same except for the peer allocation method. Merge them to reduce code duplication.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [ ]  This PR does not require documentation updates.

### CODE REVIEW

File: src/common/meta/src/ddl/table_meta.rs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    pub async fn create(
+        &self,
+        ctx: &TableMetadataAllocatorContext,
+        task: &CreateTableTask,
+    ) -> Result<TableMetadata> {
+        let table_id = self.allocate_table_id(task).await?;
+
+        let table_route = self.create_table_route(ctx, table_id, task).await?;
+
+        let region_wal_options = self.create_wal_options(&table_route)?;
```
Comment (niebayes) [inline src/common/meta/src/ddl/table_meta.rs:156]:
```suggestion
        let table_id = self.allocate_table_id(task).await?;
        let table_route = self.create_table_route(ctx, table_id, task).await?;
        let region_wal_options = self.create_wal_options(&table_route)?;
```

**Review by niebayes — APPROVED** [review]
Almost LGTM except for a non-blocking suggestion.

File: src/common/meta/src/ddl/table_meta.rs [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    }
+}
+
+pub type PeerAllocatorRef = Arc<dyn PeerAllocator>;
+
+/// [PeerAllocator] allocates [Peer]s for creating regions.
+#[async_trait]
+pub trait PeerAllocator: Send + Sync {
+    /// Allocates `regions` size [Peer]s.
+    /// Note that the returned [Peer]s do not need to be the same with `regions`.
```
Comment (WenyXu) [inline src/common/meta/src/ddl/table_meta.rs:175]:
```suggestion
```

**Review by WenyXu — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**